src/party.cpp

```diff
@@ -395,15 +395,29 @@ bool Party::setSharedExperience(Player* player, bool sharedExpActive)
 	this->sharedExpActive = sharedExpActive;
 
 	if (sharedExpActive) {
-		this->sharedExpEnabled = canEnableSharedExperience();
+		sharedExpEnabled = canEnableSharedExperience();
```

Comment:
This will change the parameter value, not the class attribute!

Reply:
I don't quite understand what you mean xD. ``canEnableSharedExperience()`` has been modified in order to set the current state of shared experience, letting us know which message to send.

Reply:
Are you sure? `canEnableSharedExperience` is a const function and does not modify the party.